cmd/snap-exec/main.go

```diff
@@ -201,8 +201,12 @@ func execApp(snapApp, revision, command string, args []string) error {
 		cmdArgs = nil
 	case "complete":
 		fullCmd[0] = defaultShell
+		completionHelper := dirs.CompletionHelperInCore
+		if info.NeedsClassic() {
+			completionHelper = filepath.Join(dirs.SnapMountDir, "core/current", dirs.CompletionHelperInCore)
```

Comment:
Hm, I think this also needs to take the "snapd" snap into account, i.e. check if there is snapd/current and if so use that. Or should we maybe just use the host system one instead of using the one coming from the snap?

Reply:
The location of the completion helpers must be dictated by the re-execution policy. We cannot just use the ones from the core or from snapd snap if the distribution has opted out of the re-exec feature.

Reply:
Tweaked this to something similar to `cmd.InternalToolPath()` but without the distro specific shenanigans, to not break when `/etc/os-release` is a file rather than a symlink (eg. on CentOS/AMZN2)